3.evaluate_model/scripts/nbconverted/class_PR_curves.py

```diff
@@ -41,52 +42,62 @@
 save_dir.mkdir(parents=True, exist_ok=True)
 
 
-# ### PR Curves for combinations of models and data subsets
+# ### Evaluate Each Model on Each Dataset
 
 # In[3]:
 
 
-fig, PR_data = class_PR_curves(train_data, log_reg_model)
-fig.suptitle("Precision Recall Curves for Final Model on Training Data")
-fig.savefig(f"{save_dir}/final_model_train_data.png")
-
-PR_data.to_csv(f"{save_dir}/final_model_train_data.tsv", sep="\t")
-print(PR_data.shape)
-PR_data.head()
-
+# directory to load the models from
+models_dir = pathlib.Path("../2.train_model/models/")
+
+# use a list to keep track of scores in tidy long format for each model and dataset combination
+compiled_class_PR_curves = []
+
+# iterate through each model (final model, shuffled baseline model, etc)
+for model_path in models_dir.iterdir():
+    model = load(model_path)
+    model_name = model_path.name.replace("log_reg_","").replace(".joblib","")
+    
+    # iterate through label datasets (labels correspond to train, test, etc)
+    # with nested for loops, we test each model on each dataset(corresponding to a label)
+    for label in data_split_indexes["label"].unique():
+        print(f"Evaluating {model_name} on dataset {label}")
+        
+        # load dataset (train, test, etc)
+        data = get_dataset(features_dataframe, data_split_indexes, label)
+        
+        fig, PR_data = class_PR_curves(data, model)
+        fig.suptitle(f"Precision Recall Curves for {model_name} on Dataset {label}")
+        plt.show()
```

Comment:
Really liked the dynamic titling here within the loop. Noticing the ability to control the figure and the similarity of output within the notebook, would it make sense to differentiate the line color (it's blue in all of them) to help the viewer distinguish each group?

Reply:
I don't think that makes sense here as there are too many classes (16) to come up with a good color split. However, in the future when we do a mixed-multiclass model (4 classes) I will definitely implement specific colors.